### PR TITLE
Lazily initialise Xcode installation status

### PR DESCRIPTION
**Pull request by cbracken**
Rather than pre-compute Xcode install path, version, and EULA status,
compute and cache on demand.

### CODE REVIEW

**Comment by cbracken** [comment]
Pre-factoring before making these calls async.

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 1]
```diff
@@ -80,54 +80,60 @@ class IMobileDevice {
...
-  bool get isInstalledAndMeetsVersionCheck => isInstalled && xcodeVersionSatisfactory;
-
-  String _xcodeSelectPath;
-  String get xcodeSelectPath => _xcodeSelectPath;
-
-  bool _isInstalled;
-  bool get isInstalled => _isInstalled;
+  bool get isInstalled {
+    if (xcodeSelectPath != null && xcodeSelectPath.isNotEmpty)
+      return false;
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/ios/mac.dart]: This should be `true`, no?

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 2]
```diff
@@ -80,54 +80,60 @@ class IMobileDevice {
...
+    if (!xcodeVersionRegex.hasMatch(xcodeVersionText))
+      return false;
+    if (!eulaSigned)
+      return false;
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/ios/mac.dart]: This doesn't match the previous logic

File: packages/flutter_tools/lib/src/ios/mac.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/mac.dart]: Whoops - tried to get clever, realised it wouldn't work and missed a revert.  Once this is in, asyncified and the process invocations extracted, we'll be able to easily add tests for these cases.

File: packages/flutter_tools/lib/src/ios/mac.dart (hunk 2 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/mac.dart]: Done.

**Review by tvolkert — APPROVED** [review]

**Pull request merged**